src/samplereader/SampleReader.h

```diff
@@ -14,6 +14,20 @@
 #include <kodi/AddonBase.h>
 #include <kodi/addon-instance/Inputstream.h>
 
+enum CryptoMode : uint16_t
+{
+  NONE,
+  AES_CTR,
+  AES_CBC
+};
```

Comment:
is preferible to have a class enum over ungrouped old enum, and with fixed values like:

```suggestion
// The crypto modes values must match android MediaCodec values
enum class CryptoMode
{
  NONE=0,
  AES_CTR=1,
  AES_CBC=2
};
```
you can also partial restore the original implementation where SetEncryptionScheme use the enum instead an integer arg variable

the only cast needed to add is in main.cpp

p->cryptoInfo->mode = static_cast<uint16_t>(cryptoInfo.m_mode);

Reply:
Sure thing, done